urlang/main.rkt

```diff
@@ -1473,12 +1473,26 @@
 
 (define (parse-object o)
   (debug (list 'parse-object (syntax->datum o)))
+  (define (duplicates? pns)
+      (remove-duplicates (map syntax-e (filter (λ(item) 
+        (>
+          (length
+            (filter (λ(other) (free-identifier=? item other)) pns))
+        1))
+      pns)
+  )))
+
+
   (with-output-language (Lur Expr)
     (syntax-parse o
       #:literal-sets (keywords)
       [(object [pn:PropertyName e] ...)
        (let ([pn (syntax->list #'(pn ...))]
              [e  (stx-map parse-expr #'(e  ...))])
+          (define dup (duplicates? pn))
+          (when (> (length dup) 0)
+            (displayln 
+              (format "Warning: object contains shadowing variables ~a: ~a" dup (syntax->datum o))))
          `(object [,pn ,e] ...))])))
```

Comment:
If `duplicates?` return `#f` in the case of no duplicates, you can write

```
(unless (duplicates pn)
   ...)
```

You have opted for displaying a warning - I am wondering whether it were better to make it an error?
That is, use something like this:

    (raise-syntax-error 'parse-object (~a "object contains shadowing variables " o) o) 
    
I can't remember, is duplicate names in objects an error in JavaScript - or is it ignored?

Reply:
Duplicate are ignored in JavaScript and shadowing do happen.